web/src/app/users/UserContactMethodList.js

```diff
@@ -98,19 +106,90 @@ export default class UserContactMethodList extends React.PureComponent {
             }
           </Config>
         </Card>
-        {this.state.edit && (
-          <UserContactMethodEditDialog
-            cmID={this.state.edit}
-            onClose={() => this.setState({ edit: null })}
-          />
-        )}
-        {this.state.delete && (
-          <UserContactMethodDeleteDialog
-            cmID={this.state.delete}
-            onClose={() => this.setState({ delete: null })}
-          />
-        )}
       </Grid>
     )
   }
 }
+
+function ReactivateButton(props) {
```

Comment:
We can do an `icon` bool, but it would result in the same amount of logic (if not more) on whether or not `IconButton` or `Button` should be rendered.

And only one dialog would ever be able to be opened at a time (would be unable to click the other button while the dialog is open), so I'm not sure I follow how this could be an issue

Reply:
I think the API of a component is more important as it's implementation. I'd rather refactor or replace a single component than have to refactor every call site with a bunch of props.

The more "flexible" the component, the more work that lives everywhere else that is using it to maintain. It's easier to replace or refactor a component that takes a string and bool, than one that takes 2 components and a string. It's also less context for the caller to only need to provide "should this be an icon?" and a contact method ID, than needing to read ReactivateButton to see what the two passed-in components need to be for it to work properly.

Reply:
I could understand where you're coming from if we were to export this component, but it's a private functional component just to save lines of code in this file, not to make something very generic for several use-cases. if we were to pass in an `icon` prop we would then have to determine what higher level component to use (`Button` vs `IconButton`) as well as what children to render (a string vs an icon), which we would then have to add additional functionality to allow an icon to be passed as a prop (which is what `buttonChild` is currently doing), instead of having it be a bool like in the first suggestion.

If this is something you feel strongly towards, I can make a generic "button" component under Util where we can specify an icon or a string to switch between button types, but personally I don't find it necessary

Reply:
That's understandable. And no, it's not a huge deal. My thinking was that we should render the dialog the same as edit and delete (and not render multiple dialogs for verify separately) to keep things consistent rather than diverging to save a few lines of code.

It might not be an issue today (referring to multiple dialogs opening), but it's more to debug tomorrow. Not having an example of how a user would get into said state, doesn't mean it won't happen either, or that it won't happen in CI tests which operate faster than users. 

So with that idea, having the dialog raised out of this component, `ReactivateButton` serves little use since it just renders whatever you pass it, and that's where the idea of moving the `icon` logic to it. My primary concern was only rendering this dialog in a different way than all the others.

It's more important to be consistent and clear than to save lines of code. Code is read far more often than it is written.

Reply:
That sounds good. I'll adjust such that the dialog logic is rendered in the parent component

Reply:
I'll just remove it.